src/ProtectedRoute.tsx

```diff
@@ -0,0 +1,33 @@
+import React from "react";
+import { Redirect, Route, RouteProps } from "react-router-dom";
+import { useKeycloak } from "@react-keycloak/web";
+
+import { hasUserRole } from "helpers/roles";
+import { STATIC_ROUTES } from "utils/routes";
+
+const ProtectedRoute = ({ ...routeProps }: RouteProps) => {
+  const { user } = { user: { roles: ["allo"], acceptedTerms: true } } as any; //useUser(); TODO
+  const { keycloak } = useKeycloak();
+  const userNeedsToLogin = !keycloak.authenticated;
+
+  if (userNeedsToLogin) {
+    return <Redirect to={STATIC_ROUTES.HOME} />;
+  }
+
+  if (!hasUserRole(user)) {
+    return <Redirect to={STATIC_ROUTES.JOIN} />;
```

Comment:
Not sure the flow with Join page and Terms and Conditions will be served by the portal

Reply:
I put it just in case, but we will remove it if its not necessary eventually